src/core/namespace.c

```diff
@@ -207,14 +206,14 @@ static const MountEntry protect_control_groups_yes_table[] = {
 };
 
 /* ProtectControlGroups=private table. Note mount_private_apivfs() always use MS_NOSUID|MS_NOEXEC|MS_NODEV so
- * flags is not set here. nsdelegate has been supported since kernels >= 4.13 so it is safe to use. */
+ * flags is not set here. */
 static const MountEntry protect_control_groups_private_table[] = {
-        { "/sys/fs/cgroup",      MOUNT_PRIVATE_CGROUP2FS, false, .read_only = false, .nosuid = true, .noexec = true, .options_const = "nsdelegate"  },
+        { "/sys/fs/cgroup",      MOUNT_PRIVATE_CGROUP2FS, false, .read_only = false },
 };
 
 /* ProtectControlGroups=strict table */
 static const MountEntry protect_control_groups_strict_table[] = {
-        { "/sys/fs/cgroup",      MOUNT_PRIVATE_CGROUP2FS, false, .read_only = true,  .nosuid = true, .noexec = true, .options_const = "nsdelegate"  },
```

Comment:
why drop nosuid/noexec? those bind mount flags, should not be dropped?

Reply:
See my comments at https://github.com/systemd/systemd/pull/36764#discussion_r1997786239

Reply:
ah, ok